### ISSUE TITLE
Sound problems

### ISSUE THREAD

**Issue by Kirkman** [dmcoles/EstyJs#2]
First, I just want to say thanks for this project. The idea of running an Atari ST in a browser is pretty awesome.

I had an idea for a blog post I could write next month with some old Atari ST Christmas demos. At first I planned to simply record videos of the demos playing inside Hatari. But then I remembered your EstyJS project, and I thought I might try it instead.

It almost works. The main problem is the music (a bunch of classic Christmas tunes). The music plays back wrong in Mac Firefox or Mac Chrome. It doesn't play at all in Safari.

[I posted an example of the problem on my website](http://breakintochat.com/files/emulators/atari-st/estyJS/). Just launch the ALITE.PRG file.

FWIW, the demos sound correct in Hatari and on my Mega STe.

**Comment by Kirkman** [comment]
I know it's been ten years, but it'd be really awesome if the sound issue could be improved. In the ten years since I noted sound issues with EstyJS, at least one other browser-based ST emulator has come out (PCEJS). But it *also* has sound problems. 

So ... I'm still looking for a good embeddable Atari ST browser that supports sound!

**Comment by kaiec** [comment]
I am not the author, but started to look into the code base and found it quite approachable. I hope we get a proper license for the code (see #6 ) to improve some things. On my wish list so far are better mouse control and this sound problem as both affect me as well.

I did not yet look into the sound implementation but would assume that there is some issue how the virtual registers for the sound chip are translated into JavaScript sound. Considering that it works overall, it might be a small fix (fingers crossed).

Long story short, if you (or someone else reading this) would be willing to help, we might be able to figure it out. Probably some good understanding of how the original hardware worked helps, my knowledge there is unfortunately very limited.
